include/odp/api/spec/dma.h

```diff
@@ -0,0 +1,522 @@
+ * @retval ODP_DMA_INVALID  DMA session could not be found
+ */
+odp_dma_t odp_dma_lookup(const char *name);
+
+/**
+ * Initialize DMA transfer parameters
+ *
+ * Initialize an odp_dma_transfer_param_t to its default values.
+ *
+ * @param param         Parameter structure to be initialized
+ */
+void odp_dma_transfer_param_init(odp_dma_transfer_param_t *param);
+
+/**
+ * Initialize DMA transfer completion parameters
+ *
+ * Initialize an odp_dma_compl_param_t to its default values.
+ *
+ * @param param         Parameter structure to be initialized
+ */
+void odp_dma_compl_param_init(odp_dma_compl_param_t *param);
+
+/**
+ * Synchronous DMA transfer operation
+ *
+ * Transfers data according to the parameters and returns when the transfer is complete.
+ *
+ * @param dma           DMA session
+ * @param transfer      Transfer parameters
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_dma_transfer_sync(odp_dma_t dma, const odp_dma_transfer_param_t *transfer);
+
+/**
+ * Start DMA transfer
+ *
+ * Start asynchronous DMA transfer operation. Transfer parameters specify data to be transferred.
+ * Completion parameters specify how transfer completion is reported.
+ *
+ * @param         dma           DMA session
+ * @param         transfer      Transfer parameters
+ * @param[in,out] completion    Transfer completion parameters
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_dma_transfer_start(odp_dma_t dma, const odp_dma_transfer_param_t *transfer,
+			   odp_dma_compl_param_t *completion);
+
+/**
+ * Start multiple DMA transfers
+ *
+ * Like odp_dma_transfer_start(), but starts 'num' transfers.
+ *
+ * @param         dma           DMA session
+ * @param         transfer      Array of transfer parameters
+ * @param[in,out] completion    Array of transfer completion parameters
+ * @param         num           Number of transfers to start. Both parameter arrays have this
+ *                              many elements.
+ *
+ * @retval Number of transfers started successfully (0 ... num)
+ * @retval <0 on failure
+ */
+int odp_dma_transfer_start_multi(odp_dma_t dma, const odp_dma_transfer_param_t *transfer[],
+				 odp_dma_compl_param_t *completion[], int num);
+
+/**
+ * Check if a DMA transfer has completed
+ *
+ * Application must call this function for every transfer that was started in poll mode until
+ * a non-zero value is returned. Transfer identifier from a previous transfer start call is used.
+ * The identifier is used as long as zero is returned. When a non-zero value is returned,
+ * the transfer has completed (either successfully or into a failure) and the identifier must not
+ * be used anymore.
+ *
+ * @param dma          DMA session
+ * @param transfer_id  Transfer identifier
+ *
+ * @retval 0  transfer has not finished
+ * @retval >0 transfer has finished successfully
+ * @retval <0 on failure
+ */
+int odp_dma_transfer_status(odp_dma_t dma, odp_dma_transfer_id_t transfer_id);
```

Comment:
If this function checks status only a single transfer? If multiple of them have been started (as a batch) by the odp_dma_transfer_start_multi(), the status function is called for each of them? If so, where it gets the transfer_id for each those transfers?

Reply:
start_multi() returns individual transfer_id for each transfer through odp_dma_compl_param_t table.